src/clj/puppetlabs/http/client/common.clj

```diff
@@ -149,3 +155,28 @@
 
 (def Response
   (schema/either NormalResponse ErrorResponse))
+
+(def OptionalMetricRegistry
+  (schema/maybe MetricRegistry))
+
+(def Metrics
+  {schema/Str Timer})
+
+(def MetricData
+  {:metric-id schema/Str
+   :count schema/Int
+   :mean schema/Num
+   :aggregate schema/Num})
+
+(def MetricsData
+  {schema/Str MetricData})
+
+(def MetricFilter
+  (schema/conditional
+   #(contains? % :url)
+   {:metric-type (schema/enum "bytes-read")
```

Comment:
Also for the clojure API, I think the metric-type enums should contain keyword values rather than string values.